pkg/azure/acr.go

```diff
@@ -53,3 +73,51 @@ func ListContainerRegistries(ctx context.Context, subscriptionId, resourceGroup
 	lgr.Debug("finished listing ACRs")
 	return acrs, nil
 }
+
+func CheckACRPullAccess(ctx context.Context, subscriptionId, resourceGroup string) error {
+	roles, err := ListRoleAssignments(ctx, subscriptionId, resourceGroup)
+	if err != nil {
+		return fmt.Errorf("listing role assignments: %w", err)
+	}
+	for _, role := range roles {
+		if (*role.Name == AcrPullRole.Name) && (*role.ID == AcrPullRole.ID) {
+			// tbarnes94: success case
+			return nil
+		}
+	}
```

Comment:
this alone does not tell us if a cluster can pull from an acr. We must check the scopes and ensure the principal id given to the is the kubelet identity.

This function must take a cluster as a parameter

Reply:
We want to check that a specific cluster has access to a specific acr